README

```diff
@@ -25,5 +26,31 @@
  - make install
 
  Make sure that you do not have libcoap-1-0-dev installed, as the headers might conflict.
+
 ===
+
+ Install TPM2 Simulator:
+ - Download code from https://sourceforge.net/projects/ibmswtpm2/
+ - Unpack into any directory
+ - cd ./src
+ - make
+ - rm -f NVChip; ./tpm_server
+
+ Install tpm2-tools:
+ - sudo apt install tpm2-tools or
+ - git clone https://github.com/tpm2-software/tpm2-tools.git
+ - cd tpm2-tools
+ - ./bootstrap
+ - ./configure --prefix=/usr
+ - make -j
+ In case if you build tpm2-tools from source check up the list of dependencies:
+ https://tpm2-tools.readthedocs.io/en/latest/INSTALL/
+
+ Install tpm2-tss package:
+ - git clone https://github.com/tpm2-software/tpm2-tss.git
+ - cd tpm2-tss
+ - ./bootstrap
+ - ./configure --prefix=/usr
+ - make -j
+ - sudo make install
```

Comment:
I followed your build instructions and I have successfully built the program but something doesn't work right:
```
ERROR:tcti:src/tss2-tcti/tcti-device.c:439:Tss2_Tcti_Device_Init() Failed to open device file /dev/tpmrm0: No such file or directory 
WARNING:tcti:src/tss2-tcti/tctildr.c:62:tcti_from_init() TCTI init for function 0x7fe3f4f0efb0 failed with a000a 
WARNING:tcti:src/tss2-tcti/tctildr.c:92:tcti_from_info() Could not initialize TCTI named: tcti-device 
ERROR:tcti:src/tss2-tcti/tctildr-dl.c:150:tcti_from_file() Could not initialize TCTI file: libtss2-tcti-device.so.0 
ERROR:tcti:src/tss2-tcti/tcti-device.c:439:Tss2_Tcti_Device_Init() Failed to open device file /dev/tpm0: No such file or directory 
WARNING:tcti:src/tss2-tcti/tctildr.c:62:tcti_from_init() TCTI init for function 0x7fe3f4f0efb0 failed with a000a 
WARNING:tcti:src/tss2-tcti/tctildr.c:92:tcti_from_info() Could not initialize TCTI named: tcti-device 
ERROR:tcti:src/tss2-tcti/tctildr-dl.c:150:tcti_from_file() Could not initialize TCTI file: libtss2-tcti-device.so.0 
WARNING:esys:src/tss2-esys/api/Esys_GetCapability.c:303:Esys_GetCapability_Finish() Received TPM Error 
ERROR:esys:src/tss2-esys/api/Esys_GetCapability.c:107:Esys_GetCapability() Esys Finish ErrorCode (0x00000100) 
Error: Esys_GetCapability()
Creating CoAP server endpoint using UDP.
Registering CoAP resources.
Entering main loop.
```

Reply:
To open "/dev/tpmrm0" you need superuser rights (sudo)

Reply:
The error is `No such file or directory ` so perhaps there is no TPM device?

Reply:
I build on HP laptop with Ubuntu where I suppose a hardware TPM is present. But I saw that it could be possible to use TPM Simulator (tmp_server) as CHARRA did. I'm just guessing that it can be configurable. If it cannot open open "/dev/tpmrm0" it try to connect to tpm_server.